src/workerd/api/basics.h

```diff
@@ -609,15 +619,54 @@ class AbortSignal final: public EventTarget {
     tracker.trackField("reason", reason);
   }
 
+  void serialize(jsg::Lock& js, jsg::Serializer& serializer);
+
+  // To test what happens if a capability is dropped before invoking release on the cloned abort
+  // signal, this method will tell every rpcClient to skip this step before destruction.
+  void skipReleaseForTest();
+
+  static jsg::Ref<AbortSignal> deserialize(
+      jsg::Lock& js, rpc::SerializationTag tag, jsg::Deserializer& deserializer);
+
+  JSG_SERIALIZABLE(rpc::SerializationTag::ABORT_SIGNAL);
+
  private:
   IoOwn<RefcountedCanceler> canceler;
   Flag flag;
   kj::Maybe<jsg::JsRef<jsg::JsValue>> reason;
   kj::Maybe<jsg::JsRef<jsg::JsValue>> onAbortHandler;
 
+  static kj::Exception abortException(
+      jsg::Lock& js, jsg::Optional<kj::OneOf<kj::Exception, jsg::JsValue>> reason);
+
   void visitForGc(jsg::GcVisitor& visitor);
 
   friend class AbortController;
+
+  /** RPC client functionality. Used if this signal was serialized. **/
+
+  // A collection of rpcClients, which will be notified if this signal is triggered and when this
+  // signal is destroyed.
+  kj::Vector<IoOwn<AbortTriggerRpcClient>> rpcClients;
+
+  // Trigger an abort on all associated clients
+  kj::Promise<void> sendToRpc(kj::Array<kj::byte>&& reason);
+
+  /** RPC server functionality. Used if this signal was deserialized. **/
+
+  // A promise that is fulfilled if an abort() message is received over RPC.
+  kj::Maybe<kj::Promise<void>> maybeRpcAbortPromise;
+
+  // A refcounted object used to receive a serialized abort reason
+  IoOwn<PendingReason> pendingReason;
```

Comment:
I think you should make this `Maybe<IoOwn<PendingReason>>`, and not set it for AbortSignals that didn't originate from RPC.

Otherwise I think if anyone today is using `getAborted` or `getReason` on an AbortSignal created by a different I/O context, they would be broken by this change. (People probably shouldn't be doing that anyway, but I could imagine someone out there is...)